packages/react-dnd-preview/src/usePreview.js

```diff
@@ -2,28 +2,48 @@ import { useDragLayer } from 'react-dnd';
 
 const getStyle = (currentOffset) => {
   const transform = `translate(${currentOffset.x}px, ${currentOffset.y}px)`;
-  return {pointerEvents: 'none', position: 'fixed', top: 0, left: 0, transform, WebkitTransform: transform};
+  return {
+    pointerEvents: 'none',
+    position: 'fixed',
+    top: 0,
+    left: 0,
+    transform,
+    WebkitTransform: transform,
+  };
+};
+
+const calculatePointerPosition = (monitor) => {
+  if (!monitor.getClientOffset() || !monitor.getInitialSourceClientOffset()) {
+    return null;
+  }
+
+  return {
+    x: monitor.getClientOffset().x - (monitor.getInitialClientOffset().x - monitor.getInitialSourceClientOffset().x),
+    y: monitor.getClientOffset().y - (monitor.getInitialClientOffset().y - monitor.getInitialSourceClientOffset().y),
+  };
 };
 
 const usePreview = () => {
   const collectedProps = useDragLayer((monitor) => {
     return {
-      currentOffset: monitor.getSourceClientOffset(),
+      currentOffset: calculatePointerPosition(monitor),
       isDragging: monitor.isDragging(),
       itemType: monitor.getItemType(),
       item: monitor.getItem(),
+      monitor,
```

Comment:
export monitor to handle these case preview change the shape when dragging, or just get more information when needed